client/state/site-settings/exporter/actions.js

```diff
@@ -83,28 +96,32 @@ export function advancedSettingsFail( siteId, error ) {
  * @param  {Number}   siteId  The ID of the site to export
  * @return {Function}         Action thunk
  */
-export function startExport( siteId ) {
-	return ( dispatch ) => {
+export function startExport( siteId, exportAll = true ) {
+	return ( dispatch, getState ) => {
 		if ( ! siteId ) {
 			return;
 		}
 
 		dispatch( {
 			type: EXPORT_START_REQUEST,
-			siteId: siteId
+			siteId,
+			exportAll,
 		} );
 
+		const state = getState();
+		const advancedSettings = exportAll ? null : prepareExportRequest( state, siteId );
+
 		const success =
 			() => dispatch( exportStarted( siteId ) );
 
 		const failure =
 			error => dispatch( exportFailed( siteId, error ) );
 
 		return wpcom.undocumented()
-			.startExport( siteId )
+			.startExport( siteId, advancedSettings )
 			.then( success )
```

Comment:
Dates are a tricky one here, because we're not really dealing with specific dates or times, but rather months. I worry that converting them to a Date object would be confusing as we'd be increasing the resolution. The date actually has a different meaning depending on whether it's used as a start date or end date. For example, the following two would mean different things:

1. **start:** '2016-06', **end:** '2016-06'

2. **start:** 2016-06-01 00:00 **end:** 2016-06-01 00:00

In the first example, we'd be including all posts in June, while in the second example no posts would be included because the dates are the same. This means we'd need to do the date calculations for 'end of month' in Calypso, which isn't necessarily a bad thing, but it feels like something the server should figure out.

Currently, the API is built to assume that the client has virtually no concept of the options it's providing. The `GET /exports/settings` endpoint provides the options in the exact same format as it expects to receive them in a `POST /exports/start`. From Calypso's perspective, it's just providing a list of options - each with a string or number value, then sending the selected values back in the same format.